src/cosmos/Swap.fixture.tsx

```diff
@@ -65,44 +98,66 @@ function Fixture() {
-      provider={connector}
-      theme={theme}
-      tokenList={tokenList}
-      width={width}
-      routerUrl={routerUrl}
-      onConnectWalletClick={() =>
-        new Promise((resolve) => {
-          console.log('integrator provided a onConnectWalletClick')
-          resolve(true) // to open our built-in wallet connect flow
-        })
-      }
-      onReviewSwapClick={() => new Promise((resolve) => resolve(true))}
-      onTokenSelectorClick={(f: Field) =>
-        new Promise((resolve) => {
-          console.log('onTokenSelectorClick', f)
-          resolve(true)
-        })
-      }
-      onTxSubmit={(txHash: string, data: any) => console.log('tx submitted:', txHash, data)}
-      onTxSuccess={(txHash: string, data: any) => console.log('tx success:', txHash, data)}
-      onTxFail={(error: Error, data: any) => console.log('tx fail:', error, data)}
-    />
+    <Row align="baseline" justify="space-around">
+      <SwapWidget
+        convenienceFee={convenienceFee}
+        convenienceFeeRecipient={convenienceFeeRecipient}
+        defaultInputTokenAddress={defaultInputToken}
+        defaultInputAmount={defaultInputAmount}
+        defaultOutputTokenAddress={defaultOutputToken}
+        defaultOutputAmount={defaultOutputAmount}
+        hideConnectionUI={hideConnectionUI}
+        locale={locale}
+        jsonRpcUrlMap={jsonRpcUrlMap}
+        defaultChainId={defaultChainId}
+        provider={connector}
+        theme={theme}
+        tokenList={tokenList}
+        width={width}
+        routerUrl={routerUrl}
+        onConnectWalletClick={() =>
+          new Promise((resolve) => {
+            addEvent({ message: 'onConnectWalletClick', data: {} })
+            resolve(true) // to open our built-in wallet connect flow
+          })
+        }
+        onReviewSwapClick={() =>
+          new Promise((resolve) => {
+            addEvent({ message: 'onReviewSwapClick', data: {} })
+            resolve(true)
+          })
+        }
+        onTokenSelectorClick={(f: Field) =>
+          new Promise((resolve) => {
+            addEvent({ message: `onTokenSelectorClick`, data: { field: f } })
+            resolve(true)
+          })
+        }
+        onTxSubmit={(txHash: string, data: any) => addEvent({ message: `onTxSubmit`, data: { ...data, txHash } })}
+        onTxSuccess={(txHash: string, data: any) => addEvent({ message: `onTxSuccess`, data: { ...data, txHash } })}
+        onTxFail={(error: Error, data: any) => addEvent({ message: `onTxFail`, data: { ...data, error } })}
+      />
+      <EventFeedWrapper>
+        <H2>Event Feed</H2>
+        {events.length > 0 && <button onClick={() => setEvents([])}>clear</button>}
+        <EventData>
+          {events?.map(({ message, data }, i) => (
+            <EventRow key={i}>
+              <div>
+                <H3 padding={0}>
+                  <Message>{message}</Message>
+                </H3>
+                <pre>{JSON.stringify(data, null, 2)}</pre>
```

Comment:
Make a new Event wrapper for the JSON:

    const Event = styled.pre`
      margin-top: 1em;
    `